stuffer/s2n_stuffer.c

```diff
@@ -437,7 +442,17 @@ int s2n_stuffer_shift(struct s2n_stuffer *stuffer)
     POSIX_ENSURE_REF(stuffer);
     struct s2n_stuffer copy = *stuffer;
     POSIX_GUARD(s2n_stuffer_rewrite(&copy));
-    uint8_t *data = stuffer->blob.data + stuffer->read_cursor;
+    /**
+     * Adding 0 to a null value is undefine behavior.
+     * This if statement prevents undefined behavior in the case
+     * where stuffer->blob.data is NULL and stuffer->read_cursor is 0.
+     */
+    uint8_t *data = stuffer->blob.data;
+
+    if (stuffer->read_cursor != 0) {
+        data += stuffer->read_cursor;
+    }
+
```

Comment:
Nit: use your whitespace wisely to improve readability.
Also, be careful about using variable names in comments. Future devs can forget to update them.
I'm also not sure the comment is necessary at all-- won't the compiler yell about NULL+0 after you start enforcing UBSAN?
```suggestion

    /**
     * Adding 0 to a null value is undefined behavior.
     * This prevents undefined behavior in the case where the read cursor is 0.
     */
    uint8_t *data = stuffer->blob.data;
    if (stuffer->read_cursor != 0) {
        data += stuffer->read_cursor;
    }

```

Reply:
I agree on the comment changes. This code is unconventional and requires some explanation for people to understand. The compiler will complain about NULL+0, but feedback won't happen until CI.